streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredVersionedKeyValueStore.java

```diff
@@ -148,13 +205,38 @@ protected <R> QueryResult<R> runRangeQuery(final Query<R> query,
             throw new UnsupportedOperationException("Versioned stores do not support RangeQuery queries at this time.");
         }
 
+        @SuppressWarnings("unchecked")
         @Override
         protected <R> QueryResult<R> runKeyQuery(final Query<R> query,
```

Comment:
Should we reserve this for `KeyQuery`, and add a new `versionedKeyQuery` method?

Reply:
Should we reserve this for `KeyQuery`, and add a new `versionedKeyQuery` method?

If you say so, I think we do not have simple key queries with versioned state stores. Therefore, we can reuse the method name. But we can follow you and change it to `versionedKeyQuery` or even have two separate methods for for each of the versioned key query types. One for single ts and one for multi ts.

Reply:
> I think we do not have simple key queries with versioned state stores

Not right now, but I think we could add support for `KeyQuery` and `RangeQuery` and target the "latest segment" only, ie `KeyQuery == VersionedKeyQuery(w/o asOfTimestamp)` in the future and should keep the door open for this.